--- proxygen/configure.ac	2018-08-20 05:26:51.000000000 +0800
+++ proxygen/configure.ac.new	2019-11-09 21:20:51.597440341 +0800
@@ -205,8 +205,8 @@
    AC_MSG_ERROR([Please install gperf first.])
 fi

-LIBS="$LIBS $BOOST_LDFLAGS -lpthread -pthread -lfolly -lglog"
-LIBS="$LIBS -ldouble-conversion -lboost_system -lboost_thread"
+LIBS="$LIBS $BOOST_LDFLAGS -lpthread -pthread"
+LIBS="$LIBS -ldouble-conversion -lboost_context -lboost_system -lboost_thread"

 AM_CONDITIONAL([HAVE_STD_THREAD], [test "$ac_cv_header_features" = "yes"])
 AM_CONDITIONAL([HAVE_X86_64], [test "$build_cpu" = "x86_64"])
